app/controllers/Controller.scala

```diff
@@ -19,29 +19,56 @@ package controllers
 import config.ViewConfig
 import connectors.des.DesConnector
 import javax.inject.{Inject, Singleton}
+import model.des.{ApprovedInformation, Transaction}
 import play.api.Logger
 import play.api.i18n.Messages
 import play.api.mvc._
+import service.des.DesService
 import uk.gov.hmrc.auth.core.{AuthorisationException, AuthorisedFunctions}
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class Controller @Inject() (mcc: MessagesControllerComponents, implicit val viewConfig: ViewConfig, page1: views.html.page1,
+class Controller @Inject() (mcc: MessagesControllerComponents, implicit val viewConfig: ViewConfig, one_payment: views.html.one_payment,
                             val authConnector: FrontendAuthConnector, errorHandler: ErrorHandler,
-                            desConnector: DesConnector, no_vat_repayments: views.html.no_vat_repayments)(implicit ec: ExecutionContext)
+                            desConnector: DesConnector, no_vat_repayments: views.html.no_vat_repayments, desService: DesService)(implicit ec: ExecutionContext)
 
   extends FrontendController(mcc) with AuthorisedFunctions {
```

Comment:
It's out of scope of this PR but at some point it would be better to get rid of `with  AuthorisedFunctions` and rely on depenency injection.

Example code from ssttp:

```
...
  @Provides
  @Singleton
  def authorisedFunctions(ac: AuthConnector): AuthorisedFunctions = new AuthorisedFunctions {
    override def authConnector: AuthConnector = ac
  }
...

package controllers.action

import com.google.inject.Inject
import config.ViewConfig
import _root_.controllers.UnhappyPathResponses
import play.api.Logger
import play.api.mvc.Results._
import play.api.mvc._
import uk.gov.hmrc.auth.core
import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
import uk.gov.hmrc.auth.core.retrieve._
import uk.gov.hmrc.auth.core.{ConfidenceLevel, Enrolments, _}
import uk.gov.hmrc.http.HeaderCarrier
import uk.gov.hmrc.play.HeaderCarrierConverter

import scala.concurrent.{ExecutionContext, Future}

final class AuthenticatedRequest[A](val request:         Request[A],
                                    val enrolments:      Enrolments,
                                    val confidenceLevel: ConfidenceLevel,
                                    val maybeUtr:        Option[String]
) extends WrappedRequest[A](request) {

  lazy val hasActiveSaEnrolment: Boolean = enrolments.enrolments.exists(_.key == "IR-SA")
}

class AuthenticatedAction @Inject() (
    af:           AuthorisedFunctions,
    viewConfig:   ViewConfig,
    badResponses: UnhappyPathResponses,
    cc:           MessagesControllerComponents)(
    implicit
    ec: ExecutionContext
) extends ActionRefiner[Request, AuthenticatedRequest] {

  import req.RequestSupport._

  override protected def refine[A](request: Request[A]): Future[Either[Result, AuthenticatedRequest[A]]] = {
    implicit val r: Request[A] = request

    af.authorised.retrieve(
      Retrievals.allEnrolments and Retrievals.confidenceLevel and Retrievals.saUtr
    ).apply {
        case enrolments ~ confidenceLevel ~ utr =>
          Future.successful(
            Right(new AuthenticatedRequest[A](request, enrolments, confidenceLevel, utr))
          )
      }
      .recover {
        case _: NoActiveSession =>
          //TODO: what is a proper value to origin
          Left(Redirect(viewConfig.loginUrl, Map("continue" -> Seq(viewConfig.frontendBaseUrl + request.uri), "origin" -> Seq("pay-online"))))
        case e: AuthorisationException =>
          Logger.debug(s"Unauthorised because of ${e.reason}, $e")
          Left(badResponses.unauthorised)
      }
  }

  override protected def executionContext: ExecutionContext = cc.executionContext

}




```

Reply:
Quite a lot of code to include when at the moment it is just using a common trait

Reply:
Done by you